test/core/client_channel/lb_policy/round_robin_test.cc

```diff
@@ -0,0 +1,251 @@
+  for (size_t i = 0; i < 20; ++i) {
+    absl::optional<std::string> address;
+    GetPickedAddress(picker.get(), address);
+    EXPECT_TRUE(address.has_value());
+    reportedUris.insert(*address);
+  }
+  EXPECT_EQ(reportedUris.size(), 3);
+  for (size_t i = 0; i < 3; ++i) {
+    EXPECT_NE(reportedUris.find(UriForSubchannel(i)), reportedUris.end())
+        << "Subchannel" << i;
+  }
+  ExpectNoStateChange();
+
+  // Shutdown channel is not considered
+  SetSubchannelConnectivityState(1, GRPC_CHANNEL_TRANSIENT_FAILURE);
+  ExpectReresolutionRequest();
+  picker = ExpectState(GRPC_CHANNEL_READY);
+  ExpectNoStateChange();
+
+  reportedUris.clear();
+  for (size_t i = 0; i < 20; ++i) {
+    absl::optional<std::string> address;
+    GetPickedAddress(picker.get(), address);
+    EXPECT_TRUE(address.has_value());
+    reportedUris.insert(*address);
+  }
+  EXPECT_EQ(reportedUris.size(), 2);
+  EXPECT_NE(reportedUris.find(UriForSubchannel(0)), reportedUris.end());
+  EXPECT_NE(reportedUris.find(UriForSubchannel(2)), reportedUris.end());
+  ExpectNoStateChange();
+}
+
+TEST_F(RoundRobinTest, OneChannelReady) {
+  auto policy = RoundRobinPolicy();
+  EXPECT_EQ(3, subchannel_pool_.size());
+  ExpectState(GRPC_CHANNEL_CONNECTING);
+
+  SetSubchannelConnectivityState(0, GRPC_CHANNEL_READY);
+  auto picker = ExpectState(GRPC_CHANNEL_READY);
+  ExpectPickComplete(picker.get(), UriForSubchannel(0));
+  ExpectNoStateChange();
+}
+
+TEST_F(RoundRobinTest, OneChannelConnecting) {
+  auto policy = RoundRobinPolicy();
+  auto picker = ExpectState(GRPC_CHANNEL_CONNECTING);
+  ExpectPickQueued(picker.get());
+  EXPECT_EQ(3, subchannel_pool_.size());
+  ExpectNoStateChange();
+}
+
+TEST_F(RoundRobinTest, OneChannelReadyToIdle) {
+  auto policy = RoundRobinPolicy();
+  ExpectState(GRPC_CHANNEL_CONNECTING);
+  SetSubchannelConnectivityState(0, GRPC_CHANNEL_READY);
+  ExpectState(GRPC_CHANNEL_READY);
+  SetSubchannelConnectivityState(0, GRPC_CHANNEL_IDLE);
+  ExpectReresolutionRequest();
+  ExpectState(GRPC_CHANNEL_CONNECTING);
+  ExpectNoStateChange();
+}
+
+TEST_F(RoundRobinTest, AllTransitFailure) {
+  auto policy = RoundRobinPolicy();
+  ExpectState(GRPC_CHANNEL_CONNECTING);
+  for (size_t i = 0; i < subchannel_pool_.size() - i; ++i) {
+    SetSubchannelConnectivityState(i, GRPC_CHANNEL_TRANSIENT_FAILURE);
+    ExpectReresolutionRequest();
+    ExpectState(GRPC_CHANNEL_CONNECTING);
+  }
+  SetSubchannelConnectivityState(subchannel_pool_.size() - 1,
+                                 GRPC_CHANNEL_TRANSIENT_FAILURE);
+  ExpectReresolutionRequest();
+  ExpectState(GRPC_CHANNEL_TRANSIENT_FAILURE);
+  ExpectNoStateChange();
+}
+
+TEST_F(RoundRobinTest, NoChannels) {
+  auto policy = MakeLbPolicy("round_robin");
+  LoadBalancingPolicy::UpdateArgs update_args;
+  update_args.addresses.emplace();
+  absl::Status status = ApplyUpdate(std::move(update_args), policy.get());
+  EXPECT_TRUE(absl::IsUnavailable(status));
```

Comment:
In this case, the LB policy will also report state TRANSIENT_FAILURE, so we should check that behavior as well.

Reply:
Done.